lib/src/model/product.dart

```diff
@@ -412,6 +414,14 @@ class Product extends JsonObject {
   @JsonKey(name: 'nutrition_grade_fr')
   String? nutriscore;
 
+  /// Map of computed Nutri-Scores for different versions (e.g. 2021, 2023).
+  @JsonKey(
+    name: 'nutriscore',
+    fromJson: NutriScore.fromJson,
+    includeToJson: false,
+  )
+  Map<NutriScoreVersion, NutriScore>? nutriscores;
```

Comment:
I would name it `nutriscoreDetails` too, as you did as a `ProductField`. That way it would be easier to keep the track.